src/libraries/System.Runtime/ref/System.Runtime.cs

```diff
@@ -7322,7 +7331,7 @@ public partial class FileStream : System.IO.Stream
         public FileStream(string path, System.IO.FileMode mode, System.IO.FileAccess access, System.IO.FileShare share, int bufferSize) { }
         public FileStream(string path, System.IO.FileMode mode, System.IO.FileAccess access, System.IO.FileShare share, int bufferSize, bool useAsync) { }
         public FileStream(string path, System.IO.FileMode mode, System.IO.FileAccess access, System.IO.FileShare share, int bufferSize, System.IO.FileOptions options) { }
-        public FileStream(string path, System.IO.FileMode mode, System.IO.FileAccess access, System.IO.FileShare share = System.IO.FileShare.Read, int bufferSize = 4096, System.IO.FileOptions options = System.IO.FileOptions.None, long allocationSize = 0) { }
+        public FileStream(string path, System.IO.FileStreamOptions options) { }
```

Comment:
If the FileStream API approved in https://github.com/dotnet/runtime/issues/45946#issuecomment-810455289 will not be needed after all, make sure to communicate that in the thread.
Also consider not closing https://github.com/dotnet/runtime/issues/45946 when this PR gets merged since there's still work to do related to the rest of other types.

Reply:
@Jozkee my plan is to just add `FileStreamOptions` overloads everywhere: https://github.com/dotnet/runtime/issues/24698#issuecomment-842884418